examples/java/plugin-auto-refresh/README.MD

```diff
@@ -0,0 +1,222 @@
+Ssh to your vCenter:
+
+```bash
+ssh root@YOUR_VC_IP
+```
+
+Then go to etc/vmware/vsphere-ui and add "allowHttp=true" to the end of the webclient.properties file.
+
+After that restart the client:
+
+```bash
+service-control --restart vsphere-ui
+```
+
+### Plugin registration
+Lastly, register the plugin to your vCenter followingin these steps:
+   1. Go to https://YOUR_VCENTER_IP/mob/?moid=ExtensionManager
+   2. Click `RegisterExtension`
+   3. Paste the extension:
+
+   ```xml
+   <extension>
+   <description>
+      <label>Remote Plug-in</label>
+      <summary>This is a sample plug-in</summary>
+   </description>
+   <key>com.vmware.sample.remote</key>
+   <company>VMware</company>
+   <version>1.0.0</version>
+   <server>
+      <url>http://PLUGIN_SERVER_IP:8089/sample-ui/plugin.json</url>
+      <description>
+      <label>Remote Plug-in</label>
+      <summary>This is a sample plug-in</summary>
+      </description>
+      <company>VMware</company>
+      <type>com.vmware.sample.remote</type>
+      <adminEmail>noreply@vmware.com</adminEmail>
+      <serverThumbprint>SERVER_THUMBPRINT</serverThumbprint>
+   </server>
+   <client>
+      <version>1.0.0</version>
+      <description>
+         <label>Remote Plug-in</label>
+         <summary>This is a sample plug-in</summary>
+      </description>
+      <company>VMware</company>
+      <type>vsphere-client-remote</type>
+      <url>http://PLUGIN_SERVER_IP:8089/sample-ui/plugin.json</url>
+   </client>
+   <lastHeartbeatTime>1970-01-01T00:00:00Z</lastHeartbeatTime>
+   </extension>
+   ```
+   
+   4. Click Invoke
+
+
+## Trigger the function
+
+Open the vSphere Client on the chassis' related hosts list page of the plugin and add a host. Whenever the host is disconnected or connected again (e.g. from another browser page) the plugin host list UI will automatically refresh with the latest list of related hosts.
+
+## Set up your own plugin auto-refresh
+
+You can embed event-driven capabilities within a UI plugin of your own. To get an idea and replicate the process for your plugin follow the steps below explaining how it is done for the remote-plugin-sample of the vSphere Client SDK.
+
+### Download the SDK plugin sample
+Firstly, you need to [download](https://code.vmware.com/web/sdk/7.0/client) the html-client sdk, version 7.0.1.0. 
+
+### Add Dockerfile and plugin-auto-refresh.yml file
+Copy the Dockerfile and the plugin-auto-refresh file from the function repo and paste them to  `html-client-sdk/samples/remote-plugin-sample/`.
+
+### Set the plugin port
+This step is optional. 
+If you want to change the port that the plugin uses, you must edit it in the server.port variable. It is located in the application.properties file, which can be found at `html-client-sdk/samples/remote-plugin-sample/target/classes`. Also, the endpoint in the upstream_url environmental variable must be updated with the new port in the Dockerfile.
+
+### Allow http connection
```

Comment:
We should leave a comment that this might cause security concerns/issues and is not recommended for production.

Reply:
OpenFaaS works with http connection which is acceptable in most cases as only OpenFaaS accesses the pods. However, in case of a plugin running in a container, we need to access it from outside, as well as through OpenFaaS. Then the http connection is not a good solution, but this is an OpenFaaS related issue.